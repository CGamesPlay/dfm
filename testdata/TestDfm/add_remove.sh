#!/bin/bash
# Tests that specific-path commands are working as intended.
set -e
. "$(dirname "$0")/../helpers.sh"

export HOME="$(pwd)/home"
export DFM_DIR="$HOME/dfmdir"

mkdir -p ~/dfmdir/files ~/.config/bash
echo 'config file' > ~/.bashrc
echo 'config file' > ~/.config/bash/00-test.sh
echo 'autoclean' > ~/dfmdir/files/AUTOCLEAN

dfm add ~/.bashrc && fail 'dfm add without init allowed'

dfm init --repos files
dfm link
rm ~/dfmdir/files/AUTOCLEAN
[ -L ~/AUTOCLEAN ] || fail "AUTOCLEAN is missing"

banner 'Importing bash config'
cd ~/.config/bash
dfm add ~/.bashrc .
[ -L ~/.bashrc ] || fail "bashrc is missing"
[ -L 00-test.sh ] || fail "00-test is missing"

banner 'Importing a new config file'
echo 'new config' > 10-test.sh
mv 10-test.sh "$DFM_DIR/files/.config/bash/"
dfm link 10-test.sh
[ -e 10-test.sh ] || fail "10-test is missing"
[ -L ~/AUTOCLEAN ] || fail "AUTOCLEAN is missing"

banner 'Reversing the import'
dfm remove 10-test.sh
mv "$DFM_DIR/files/.config/bash/10-test.sh" .
[ -e 10-test.sh ] || fail "10-test is missing"
[ -L ~/AUTOCLEAN ] || fail "AUTOCLEAN is missing"

banner 'Running autoclean'
echo 'new remote config' > "$DFM_DIR/files/.config/bash/20-test.sh"
dfm link
[ -e 10-test.sh ] || fail "10-test is missing"
[ -e 20-test.sh ] || fail "20-test is missing"
[ ! -L ~/AUTOCLEAN ] || fail "AUTOCLEAN is present"
