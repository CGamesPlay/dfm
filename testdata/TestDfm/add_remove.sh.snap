$ dfm init --repos files
Initialized /test/home/dfmdir as a dfm directory.
$ dfm link
files/AUTOCLEAN -> /test/home/AUTOCLEAN

# Importing bash config
$ dfm add /test/home/.bashrc .
added .bashrc
added .config/bash/00-test.sh

# Importing a new config file
$ dfm link 10-test.sh
files/.config/bash/10-test.sh -> /test/home/.config/bash/10-test.sh

# Reversing the import
$ dfm remove 10-test.sh
removed .config/bash/10-test.sh

# Running autoclean
$ dfm link
files/.config/bash/20-test.sh -> /test/home/.config/bash/20-test.sh
removed AUTOCLEAN
